stdlib/public/core/Mirror.swift

```diff
@@ -499,67 +499,70 @@ extension Mirror {
 /// Dictionary Literals as Function Parameters
 /// ------------------------------------------
 ///
-/// When calling a function with a `DictionaryLiteral` parameter, you can pass
+/// When calling a function with a `KeyValuePairs` parameter, you can pass
 /// a Swift dictionary literal without causing a `Dictionary` to be created.
 /// This capability can be especially important when the order of elements in
 /// the literal is significant.
 ///
 /// For example, you could create an `IntPairs` structure that holds a list of
 /// two-integer tuples and use an initializer that accepts a
-/// `DictionaryLiteral` instance.
+/// `KeyValuePairs` instance.
 ///
 ///     struct IntPairs {
 ///         var elements: [(Int, Int)]
 ///
-///         init(_ elements: DictionaryLiteral<Int, Int>) {
+///         init(_ elements: KeyValuePairs<Int, Int>) {
 ///             self.elements = Array(elements)
 ///         }
 ///     }
 ///
 /// When you're ready to create a new `IntPairs` instance, use a dictionary
 /// literal as the parameter to the `IntPairs` initializer. The
-/// `DictionaryLiteral` instance preserves the order of the elements as
+/// `KeyValuePairs` instance preserves the order of the elements as
 /// passed.
 ///
 ///     let pairs = IntPairs([1: 2, 1: 1, 3: 4, 2: 1])
 ///     print(pairs.elements)
 ///     // Prints "[(1, 2), (1, 1), (3, 4), (2, 1)]"
 @_fixed_layout // FIXME(sil-serialize-all)
-public struct DictionaryLiteral<Key, Value> : ExpressibleByDictionaryLiteral {
-  /// Creates a new `DictionaryLiteral` instance from the given dictionary
+public struct KeyValuePairs<Key, Value> : ExpressibleByDictionaryLiteral {
+  /// Creates a new `KeyValuePairs` instance from the given dictionary
   /// literal.
   ///
   /// The order of the key-value pairs is kept intact in the resulting
-  /// `DictionaryLiteral` instance.
+  /// `KeyValuePairs` instance.
   public init(dictionaryLiteral elements: (Key, Value)...) {
     self._elements = elements
   }
   @usableFromInline // FIXME(sil-serialize-all)
   internal let _elements: [(Key, Value)]
 }
 
-/// `Collection` conformance that allows `DictionaryLiteral` to
+@available(swift, deprecated: 5.0, renamed: "KeyValuePairs")
+public typealias DictionaryLiteral = KeyValuePairs
```

Comment:
Similar to #17711, should this be a generic typealias?

```diff
 @available(swift, deprecated: 5.0, renamed: "KeyValuePairs")
-public typealias DictionaryLiteral = KeyValuePairs
+public typealias DictionaryLiteral<Key, Value> = KeyValuePairs<Key, Value>
```

Reply:
I don't have a preference either way. @benrimmington you have write access to the branch. Please feel free to update it.

Reply:
@masters3d I've rebased and tested the changes locally, but I can't push to your branch. Please try adding me as a collaborator again.

Reply:
@masters3d I'm probably doing something wrong, but authentication fails on the command line (`git push masters3d +renamed-to-KeyValueArray`).

I also tried pushing from Xcode, but it fails because your branch is 1105 commits behind `apple:master`. If you can rebase your branch from upstream, I'll try again.

Reply:
https://help.github.com/articles/pushing-to-a-remote/![image](https://user-images.githubusercontent.com/6539412/43377191-40e3c51a-9374-11e8-8afd-9ca18412d848.png)

You should be able to push, try pushing a branch with a new name (that should work)

Reply:
@masters3d If I create a new branch in your repository, it won't update this pull request. I've pushed to [`benrimmington:renamed-to-KeyValueArray`](https://github.com/benrimmington/swift/tree/renamed-to-KeyValueArray) for now, but it will be simpler to ignore that and make your own changes instead.